vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ExternalLinkManagerTest.java

```diff
@@ -0,0 +1,240 @@
+        IExternalLink externalLink = new ExternalLink();
+        externalLink.setId("EXL001");
+        externalLink.setSpace(space);
+        externalLinkDisplay.setExternalLink(externalLink);
+        externalLinkDisplayList.add(externalLinkDisplay);
+        externalLinkDisplay = new ExternalLinkDisplay();
+        externalLink = new ExternalLink();
+        externalLink.setId("EXL002");
+        externalLink.setSpace(space);
+        externalLinkDisplay.setExternalLink(externalLink);
+        externalLinkDisplayList.add(externalLinkDisplay);
+        List<IExternalLinkDisplay> externalLinkObj = externalLinkDisplayList;
+        Mockito.when(externalLinkDisplayRepo.findExternalLinkDisplaysForSpace(space.getId())).thenReturn(externalLinkObj);
+        managerToTest.deleteLinkDisplayRepo(externalLink);
+        Mockito.verify(externalLinkDisplayRepo).deleteByExternalLink(externalLink);
+    }
+
+    @Test
+    public void test_deleteLinkRepo_linkPresent(){
+        ISpace space = new Space();
+        space.setId("SPA001");
+        ExternalLink externalLink = new ExternalLink();
+        externalLink.setId("EXL001");
+        externalLink.setSpace(space);
+        Mockito.when(externalLinkRepo.save(externalLink)).thenReturn(externalLink);
+        managerToTest.deleteLinkRepo(externalLink);
+        Mockito.verify(externalLinkRepo).delete(externalLink);
+    }
+
+    @Test
+    public void test_updateDisplayLink_success() throws SpaceDoesNotExistException, LinkDoesNotExistsException, ImageCouldNotBeStoredException {
+        ISpace space = new Space();
+        space.setId(spaceId1);
+        ExternalLinkDisplay externalLinkDisplay = new ExternalLinkDisplay();
+        IExternalLink externalLink = new ExternalLink();
+        externalLink.setId("EXL001");
+        externalLink.setSpace(space);
+        externalLink.setTarget(new ExternalLinkValue("stackOverflow.com"));
+        externalLinkDisplay.setId("EXLD001");
+
+        externalLinkDisplay.setExternalLink(externalLink);
+
+        externalLinkDisplay.setName("TestExternalEdited");
+        externalLinkDisplay.setPositionX(100);
+        externalLinkDisplay.setPositionY(300);
+        externalLinkDisplay.setType(DisplayType.ALERT);
+
+        Mockito.when(spaceManager.getSpace(spaceId1)).thenReturn(space);
+        ExternalLink newExternalLink = new ExternalLink(); 
+        newExternalLink.setId("externalLink1");
+        Optional<ExternalLink> mockExternalLink = Optional.of((ExternalLink)externalLink);
+        Mockito.when(externalLinkRepo.findById(externalLink.getId())).thenReturn(mockExternalLink);
+
+        Optional<ExternalLinkDisplay> mockExternalLinkDisplay = Optional.of((ExternalLinkDisplay)externalLinkDisplay);
+        Mockito.when(externalLinkDisplayRepo.findById(externalLinkDisplay.getId())).thenReturn(mockExternalLinkDisplay);
+
+        Mockito.when(managerToTest.updateLinkAndDisplay(externalLink, externalLinkDisplay)).thenReturn(externalLinkDisplay);
```

Comment:
whenever you write a mock for `managerToTest` you're probably doing something wrong.